### PR TITLE
NMM Appveyor Support

### PR DESCRIPTION
**Pull request by spbennett**
NMM Appveyor Support
- Build badging added to README.md to show build status of
latest pushes.
- Onboarded to Appveyor build system.

Read more here:
https://www.appveyor.com

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
 # Nexus Mod Manager
+[![Build status](https://ci.appveyor.com/api/projects/status/mgt1kuc1a0eqju7m?svg=true)](https://ci.appveyor.com/project/Nexus-Mods/nexus-mod-manager)
```
Comment (spbennett) [inline README.md]: API key should be replaced with token from official Nexus-Mods appveyor account.

**Comment by squid-box** [comment]
This would resolve #430.

Suggested changes (as seen in #430):
#### Version 
Change version to `{build}` instead of `1.0.{build}`.

A separate NMM improvement would be to sync the actual version of NMM with the build numbers, but that requires a little bit more work than just changing the appveyor.yml file.

#### Installer
Use the fact that Appveyor images come with Inno Setup installed, and build the installer - useful for testing (and since there's no valid certificate there's no use in building it somewhere else at the moment). 

Add a powershell script to `after_build`:
```
after_build:
- ps: "$workingDirectory = [System.Environment]::ExpandEnvironmentVariables(\"%APPVEYOR_BUILD_FOLDER%\") \nStart-Process -FilePath \"C:\\Program Files (x86)\\Inno Setup 5\\Compil32.exe\" -ArgumentList (\"/cc `\"Setup\\setup.iss`\"\") -WorkingDirectory $workingDirectory -Wait"
```

Then add an artifact publishing step:

```
artifacts:
- path: Stage\Installer\*.exe
  name: Installer
```

This also requires us to explicitly build against Release, so you could explicitly set this:
```
configuration: Release
platform: Any CPU
```

**Comment by spbennett** [comment]
Thanks for the comments Joel.  I couldn't get the installer artifact building to complete.  It ran for ~20 minutes before I killed the appveyor vm.

I'm on OSX, so I'll admit I don't have a good test bed for debugging the artifact generation code.

**Comment by squid-box** [comment]
Ah right, I think I know why: the installer build fails and Inno Setup just shows an error message waiting for input. (If I had set this up from the beginning, I would not have gone with Inno Setup :) ...)

You could try changing the following line to **false** in your branch, and it should build fine: [/Setup/setup.iss:40](https://github.com/Nexus-Mods/Nexus-Mod-Manager/blob/master/Setup/setup.iss#L40) (at least that's what happened to me when I tried this a while back)

**Comment by squid-box** [comment]
@spbennett: I take it my suggestion didn't work? Too bad :(

**Comment by spbennett** [comment]
Unfortunately it made no apparent change to my build:
https://ci.appveyor.com/project/spbennett/nexus-mod-manager/build/18-master

I think it would still be valuable to get a barebones Appveyor CI build running, and we could always tackle installer build integration afterward ☺️

**Comment by squid-box** [comment]
Agreed, I won't push further for it at this time :)

**Pull request closed without merging**